Doc/c-api/hash.rst

```diff
@@ -17,6 +17,29 @@ See also the :c:member:`PyTypeObject.tp_hash` member.
 
    .. versionadded:: 3.2
 
+.. c:macro:: PyHASH_MODULUS
+
+   The Mersenne prime ``P = 2**n -1``, used for numeric hash scheme.
```

Comment:
I'm not sure that everybody knows about it. I suggest adding a link to Wikipedia:

```suggestion
   The `Mersenne prime <https://en.wikipedia.org/wiki/Mersenne_prime>`_ ``P = 2**n -1``, used for numeric hash scheme.
```